hardware_drivers/navigator_kill_board/nodes/kill_board_driver.py

```diff
@@ -81,12 +97,51 @@ def run(self):
     def update_ros(self):
         self.update_hw_kill()
         self.publish_diagnostics()
+        if self.ctrl_msg_received is True:
+            self.publish_joy()
+            self.ctrl_msg_received = False
 
     def handle_byte(self, msg):
         '''
         React to a byte recieved from the board. This could by an async update of a kill status or
         a known response to a recent request
         '''
+        # If the controller message start byte is received, next 8 bytes are the controller data
+        if msg == constants['CONTROLLER']:
+            self.ctrl_msg_count = 8
+            self.ctrl_msg_timeout = rospy.Time.now()
+            return
+        # If receiving the controller message, record the byte as stick/button data
+        if (self.ctrl_msg_count > 0) and (self.ctrl_msg_count <= 8):
+            # If 1 second has passed since the message began, timeout and report warning
+            if (rospy.Time.now() - self.ctrl_msg_timeout) >= rospy.Duration(1):
+                self.ctrl_msg_received = False
+                self.ctrl_msg_count = 0
+                rospy.logwarn('Timeout receiving controller message. Please disconnect controller.')
+            if self.ctrl_msg_count > 2:  # The first 6 bytes in the message are stick data bytes
```

Comment:
If it's the first 6 bytes, why is this >2? Is it actually the last 6 bytes?

Reply:
When the controller message start byte (\xA0) is received, the ctrl_msg_count is set to 8 to indicate how many bytes are remaining, and decreases. If the count is more than 2, it's not the last 2 bytes yet.